packages/treeview/src/TreeviewContainer.spec.tsx

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
```

Comment:
### Test alignment

I think the structure of the tests could be updated to align more with `react-testing-library`. The majority of tests should be tested from the user's perspective.

* Tests retrieve DOM nodes using [priority](https://testing-library.com/docs/queries/about/#priority) querying.
* Tests simulate [user events](https://testing-library.com/docs/ecosystem-user-event/) on DOM nodes
* Tests assert against DOM (what the user would experience)

### Test setup

I think the test components could be simplified, without the usage of container or render props. Each test could render something like:

```js
const Treeview = () => {
 const { ...stuff } = useTreeview();
  return <ul>...</ul>;
};

test('the last treelist item expands when right arrow is pressed', () => {
  render(<Treeview />)
  
  // Simulate user events to get focus to last treeitem

  // Simulate user event to press right arrow key

  // Assert against the DOM for expected changes
})
```

Also, I think the tests can be organized into three sections:

1) Shared behavior 
2) Uncontrolled usage 
3) Controlled usage

So the structure could look like:

```js
describe('useTreeview', () => {
  // shared tests go here

  describe('uncontrolled usages', () => {
    // tests 
  });

  describe('controlled usages', () => {
    // tests 
  });
});
```

### Test coverage

Although this could wait until there's more confidence in the API, I thought I would bring to attention that new hooks will need to have close to 100% test coverage. Also, I think this could help provide some insight on how the current API is used, and how it could be improved.

### Test example

Feel free to take a look at [GridContainer.spec.tsx](https://github.com/zendeskgarden/react-containers/blob/main/packages/grid/src/GridContainer.spec.tsx) which demonstrates the thoughts from above.

Reply:
Done!